packages/application-components/src/components/custom-views/custom-view-loader/custom-view-loader.tsx

```diff
@@ -0,0 +1,142 @@
+  url: string;
+  type: 'custom-panel' | 'custom-tab';
+  typeConfig?: {
+    size?: 'small' | 'large';
+  };
+  locators: string[];
+};
+
+type TCustomViewLoaderProps = {
+  customView: TCustomView;
+  onClose: () => void;
+};
+
+const isIframeReady = (iFrameElementRef: HTMLIFrameElement) => {
+  try {
+    return iFrameElementRef?.contentWindow?.document.readyState === 'complete';
+  } catch {
+    // Trying to access the contentWindow of a cross-origin iFrame will throw an error.
+    // We are not supposed to even get here because the iFrame must use
+    // a URL from our very same domain (the custom view is proxied through our http-proxy service).
+    return false;
+  }
+};
+
+const CustomPanelIframe = styled.iframe`
+  height: 100%;
+  width: 100%;
+  border: none;
+`;
+
+function CustomViewLoader(props: TCustomViewLoaderProps) {
+  const iFrameElementRef = useRef<HTMLIFrameElement>(null);
+  const appContext = useApplicationContext();
+  const intercomChannel = useRef(new MessageChannel());
+  const showNotification = useShowNotification();
+  const intl = useIntl();
+
+  const messageFromIFrameHandler = useCallback((event: MessageEvent) => {
+    if (event.data.origin === window.location.origin) {
+      console.log('message received from iframe port: ', event);
+    }
+  }, []);
+
+  // onLoad handler is called from the iFrame even where the URL is not valid
+  // (blocked by CORS, 404, etc.) so we need to make sure the iFrame is ready
+  const onLoadSuccessHandler = useCallback(() => {
+    // Show error and block if the iFrame is not ready
+    // (error loading it)
+    if (!isIframeReady(iFrameElementRef.current!)) {
+      showNotification({
+        domain: DOMAINS.PAGE,
+        kind: NOTIFICATION_KINDS_PAGE.error,
+        text: intl.formatMessage(messages.loadError),
+      });
+      return;
+    }
+
+    // Listen for messages from the iFrame
+    intercomChannel.current.port1.onmessage = messageFromIFrameHandler;
+
+    // Transfer port2 to the iFrame so it can send messages back privately
+    iFrameElementRef.current?.contentWindow?.postMessage(
+      'mc_init',
+      window.location.href,
+      [intercomChannel.current.port2]
+    );
+
+    // Send the initialization message to the iFrame
+    intercomChannel.current.port1.postMessage({
+      source: 'mc-host-application',
+      destination: `custom-view-${props.customView.id}`,
+      eventName: 'hostAcknowledgesIframeLoaded',
+      eventData: {
+        appContext,
+      },
+    } as TCustomViewIframeMessage);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // Currently we only support custom panels
+  if (props.customView.type !== 'custom-panel') {
+    reportErrorToSentry(
+      new Error(
+        `CustomViewLoader: Provided Custom View has an unsupported type: ${props.customView.type}`
+      )
+    );
+    return null;
+  }
+
+  return (
+    <CustomPanel
+      title={`Custom View: ${props.customView.defaultLabel}`}
```

Comment:
Is the prefix needed?

Reply:
Since titles should be user friendly, I think it's worth it to have the prefix so readers can know what the iFrame is about.